internal/controller/cluster_upgrade.go

```diff
@@ -495,10 +497,14 @@ func checkPodInitContainerIsOutdated(pod *corev1.Pod, _ *apiv1.Cluster) (rollout
 }
 
 func checkHasMissingPVCs(pod *corev1.Pod, cluster *apiv1.Cluster) (rollout, error) {
-	if persistentvolumeclaim.InstanceHasMissingMounts(cluster, pod) {
+	if missingPVC, missingTbsPVC := persistentvolumeclaim.InstanceHasMissingMounts(
+		cluster,
+		pod,
+	); missingPVC {
 		return rollout{
-			required:             true,
-			primaryForceRecreate: true,
+			required: true,
+			// we force the recreation of the primary pod if the tablespace PVC is missing
+			primaryForceRecreate: missingTbsPVC,
```

Comment:
I'm a bit confused here.
So, if there is a missing tablespace, we force an in-place primary upgrade, and otherwise we go with the primaryUpdate strategy.
But, when looking for `InstanceHasMissingMounts`, we iterate over mounts. We could find that there is both a PG WAL and a table space needing creation. Why will the TBS "win"?

Reply:
@jsilvela For a missing wal volume, the old primary is able to restart postgres and demote itself without issue, because wal folder can locates under pg_data. For missing tablespace volume, the old primary can not restart postgres (crashloop )as new primary (through switchover) already have tablespace created.

so for missing wal, old primary restart and demoting, and then podSpec recreate the pod (pod restart)
for missing tablespace, old primary should recreate pod directly without a switchover, otherwize, switchover will failed on postgres restart.